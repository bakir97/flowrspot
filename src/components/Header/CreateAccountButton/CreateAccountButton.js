import React from "react";
import styles from "./CreateAccountButton.module.scss";

const CreateAccountButton = ({ openModal, showMenu, toggleMenu }) => {
  const triggerOpenModal = () => {
    window.document.body.style.overflow = "hidden";
    showMenu && toggleMenu();
    openModal("createAccount");
  };
  return (
    <button onClick={triggerOpenModal} className={styles.button}>
      New Account
    </button>
  );
};

export default CreateAccountButton;
