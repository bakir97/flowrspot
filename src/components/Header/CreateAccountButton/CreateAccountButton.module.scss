@import "../../../assets/global/styles/mixins.scss";

.button {
  @include rounded-orange-button;
}
