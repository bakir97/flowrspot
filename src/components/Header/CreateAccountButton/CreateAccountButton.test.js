import React from "react";
import { shallow } from "enzyme";
import CreateAccountButton from "./CreateAccountButton";
describe("CreateAccountButton", () => {
  it("should render correctly", () => {
    const component = shallow(<CreateAccountButton />);
    expect(component).toMatchSnapshot();
  });
  it("should call trigger open modal and toggle menu ", () => {
    const mockFn = jest.fn();
    const component = shallow(
      <CreateAccountButton openModal={mockFn} toggleMenu={mockFn} showMenu />
    );
    component.find("button").simulate("click");
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});
