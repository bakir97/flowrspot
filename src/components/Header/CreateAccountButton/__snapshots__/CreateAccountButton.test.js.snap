// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountButton should render correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  New Account
</button>
`;
